pkgs/by-name/ed/edopro/package.nix

```
@@ -0,0 +1,299 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  fetchzip,
+  premake5,
+  writeShellScriptBin,
+  runCommandLocal,
+  symlinkJoin,
+  imagemagick,
+  bzip2,
+  curl,
+  flac,
+  fmt_9,
+  freetype,
+  libevent,
+  libgit2,
+  libGL,
+  libGLU,
+  libjpeg,
+  libpng,
+  libvorbis,
+  libX11,
+  libxkbcommon,
+  libXxf86vm,
+  lua5_3,
+  mono,
+  nlohmann_json,
+  openal,
+  SDL2,
+  sqlite,
+  wayland,
+  egl-wayland,
+  covers_url ? "https://pics.projectignis.org:2096/pics/cover/{}.jpg",
+  fields_url ? "https://pics.projectignis.org:2096/pics/field/{}.png",
+  # While ygoprodeck has higher quality images, "spamming" of their api results in a ban.
+  # Thats why this link can change since it's compiled into the program, However it will
+  # download assets when needed so it is unlikely to get banned. Unless you search the
+  # card list with no filters of any kind. When testing use ProjectIgnis' website instead.
+  pics_url ? "https://images.ygoprodeck.com/images/cards/{}.jpg",
+}:
+let
+  deps = import ./deps.nix;
+in
+let
+  assets = {
+    pname = "assets";
+    version = deps.edopro-version;
+
+    src = fetchzip {
+      url = "https://github.com/ProjectIgnis/edopro-assets/releases/download/${deps.edopro-version}/ProjectIgnis-EDOPro-${deps.edopro-version}-linux.tar.gz";
+      sha256 = deps.assets-hash;
+    };
+  };
+
+  irrlicht-edopro = stdenv.mkDerivation {
+    pname = "irrlicht-edopro";
+    version = "1.9-custom";
+
+    src = fetchFromGitHub {
+      owner = "edo9300";
+      repo = "irrlicht1-8-4";
+      rev = deps.irrlicht-rev;
+      hash = deps.irrlicht-hash;
+    };
+
+    buildInputs = [
+      libGLU
+      libX11
+      libxkbcommon
+      libXxf86vm
+      wayland
+    ];
+
+    enableParallelBuilding = true;
+    buildFlags = [
+      "NDEBUG=1"
+    ];
+    makeFlags = [
+      "-C"
+      "source/Irrlicht"
+    ];
+
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out/{bin,include}
+      cp lib/Linux/libIrrlicht.a $out/bin
+      cp -r include/* $out/include
+
+      runHook postInstall
+    '';
+  };
+
+  ocgcore = stdenv.mkDerivation rec {
+    pname = "ygopro-core";
+    version = deps.ocgcore-rev;
+
+    src = fetchFromGitHub {
+      owner = "edo9300";
+      repo = "ygopro-core";
+      rev = version;
+      hash = deps.ocgcore-hash;
+      fetchSubmodules = true;
+    };
+
```

[ocgcore-lua-symbols.patch.txt](https://github.com/user-attachments/files/17463588/ocgcore-lua-symbols.patch.txt)


```suggestion

    patches = [
      ./ocgcore-lua-symbols.patch
    ];

```

---

Add the uploaded file as well, after renaming it to the name in the suggestion. (GitHub doesn't let you upload .patch files >.>)

---

Is the postpatch not enough for this?

---

I didn't include the `postPatch` from the discussion on Discord because that was too hacky, doing those fixes via a patch is better.